tests/test_latlon.py

```diff
@@ -0,0 +1,166 @@
+# -*- coding: utf-8 -*-
+"""
+This is the unittest for latlon related routines.
+"""
+
+import numpy as np
+import unittest
+import gstools as gs
+
+
+def _rel_err(a, b):
+    return np.abs(a / ((a + b) / 2) - 1)
+
+
+class ErrMod(gs.CovModel):
+    def cor(self, h):
+        return np.exp(-(h ** 2))
+
+    def fix_dim(self):
+        return 2
+
+
+class TestCondition(unittest.TestCase):
+    def setUp(self):
+        self.cmod = gs.Gaussian(
+            latlon=True, var=2, len_scale=777, rescale=gs.EARTH_RADIUS
+        )
+        self.lat = self.lon = range(-80, 81)
+
+        self.data = np.array(
+            [
+                [52.9336, 8.237, 15.7],
+                [48.6159, 13.0506, 13.9],
+                [52.4853, 7.9126, 15.1],
+                [50.7446, 9.345, 17.0],
+                [52.9437, 12.8518, 21.9],
+                [53.8633, 8.1275, 11.9],
+                [47.8342, 10.8667, 11.4],
+                [51.0881, 12.9326, 17.2],
+                [48.406, 11.3117, 12.9],
+                [49.7273, 8.1164, 17.2],
+                [49.4691, 11.8546, 13.4],
+                [48.0197, 12.2925, 13.9],
+                [50.4237, 7.4202, 18.1],
+                [53.0316, 13.9908, 21.3],
+                [53.8412, 13.6846, 21.3],
+                [54.6792, 13.4343, 17.4],
+                [49.9694, 9.9114, 18.6],
+                [51.3745, 11.292, 20.2],
+                [47.8774, 11.3643, 12.7],
+                [50.5908, 12.7139, 15.8],
+            ]
+        )
+
+    def test_conv(self):
+        p_ll = gs.tools.geometric.latlon2pos((self.lat, self.lon), 2.56)
+        ll_p = gs.tools.geometric.pos2latlon(p_ll, 2.56)
+        for i, v in enumerate(self.lat):
+            self.assertAlmostEqual(v, ll_p[0, i])
+            self.assertAlmostEqual(v, ll_p[1, i])
+        self.assertAlmostEqual(
+            8, self.cmod.anisometrize(self.cmod.isometrize((8, 6)))[0, 0]
+        )
+        self.assertAlmostEqual(
+            6, self.cmod.anisometrize(self.cmod.isometrize((8, 6)))[1, 0]
+        )
+        self.assertAlmostEqual(
+            1, self.cmod.isometrize(self.cmod.anisometrize((1, 0, 0)))[0, 0]
+        )
+
+    def test_cov_model(self):
+        self.assertAlmostEqual(
+            self.cmod.vario_yadrenko(1.234),
+            self.cmod.sill - self.cmod.cov_yadrenko(1.234),
+        )
+        self.assertAlmostEqual(
+            self.cmod.cov_yadrenko(1.234),
+            self.cmod.var * self.cmod.cor_yadrenko(1.234),
+        )
+        # test if correctly handling tries to set anisotropy
+        self.cmod.anis = [1, 2]
+        self.cmod.angles = [1, 2, 3]
+        self.assertAlmostEqual(self.cmod.anis[0], 1)
+        self.assertAlmostEqual(self.cmod.anis[1], 1)
+        self.assertAlmostEqual(self.cmod.angles[0], 0)
+        self.assertAlmostEqual(self.cmod.angles[1], 0)
+        self.assertAlmostEqual(self.cmod.angles[2], 0)
+
+    def test_vario_est(self):
+
```

Comment:
Is this black?!?!

Reply:
Black allows one blank line below function definitions. I can remove it.